dateparser/freshness_date_parser.py

```diff
@@ -13,7 +16,40 @@ class FreshnessDateDataParser(object):
     def __init__(self, now=None):
         self.now = now or datetime.utcnow()
 
+    def _are_all_words_units(self, date_string):
+        skip = [_UNITS,
+                r'about|ago|\d+',
+                r':|[ap]m'] 
+
+        date_string = re.sub(r'\s+', ' ', date_string.strip())
+
+        words = filter(lambda x: x if x else False, re.split('\W', date_string))
+        words = filter(lambda x: not re.match(r'%s' % '|'.join(skip), x), words)
+        return not bool(words)
+
+    def _parse_time(self, date_string):
+        date_string = re.sub(r'\d+\s*(%s)' % _UNITS, '', date_string)
```

Comment:
May be we should use regex and flags identical to the one in `get_kwargs`? How about creating a compiled one and use in both places?